### ISSUE TITLE
Unable to connect to mastodon instances with signToActivityPubGet enabled

### ISSUE THREAD

**Issue by georgianachao** [syuilo/Misskey#8723]
I just installed misskey on my server and found that I couldn't interact with users on mastodon instances that had turned on secure mode. I have looked at some discussions on this issue and have enabled `signToActivityPubGet` in `.config/default.yml` . However, it still doesn't work. Is there any approach to solve this?

**Comment by Johann150** [comment]
Have you restarted Misskey after changing the configuration? Otherwise the new configuration will not take effect.

**Comment by georgianachao** [comment]
> Have you restarted Misskey after changing the configuration? Otherwise the new configuration will not take effect.

yes! And I actually configured my instance with `signToActivityPubGet` enabled when I first installed misskey (I’m a mastodon user and I knew this issue before I installed misskey and actually came into it). Thanks for the reply.

**Comment by Johann150** [comment]
Can you check if there are any errors in the logfile?

**Comment by georgianachao** [comment]
> Can you check if there are any errors in the logfile?

Here are the errors in the logfile that occurred when i was trying to follow some remote users on mastodon instances that enabled secure mode.
 
```
May 23 18:33:34 vultr misskey[8307]: INFO 1        [remote ap]        Creating the Person: https://onlycasino.legal/users/blackjack
May 23 18:33:34 vultr misskey[8307]: ERR  1        [remote ap]        QueryFailedError: character with byte sequence 0xf0 0x9f 0x8e 0xb5 in encoding "UTF8" has no equivalent in encoding "LATIN1"
May 23 18:33:34 vultr misskey[8307]: ERR  1        [api]        Internal error occurred in ap/show: character with byte sequence 0xf0 0x9f 0x8e 0xb5 in encoding "UTF8" has no equivalent in encoding "LATIN1"
May 23 18:33:34 vultr misskey[8307]: WARN 1        [api]        SLOW API CALL DETECTED: ap/show (10341.942075000145ms)
May 23 18:34:17 vultr misskey[8307]: INFO 1        [remote ap]        Creating the Person: https://boilingcong.ee/users/auis
May 23 18:34:17 vultr misskey[8307]: ERR  1        [remote ap]        QueryFailedError: character with byte sequence 0xe5 0xae 0xb6 in encoding "UTF8" has no equivalent in encoding "LATIN1"
May 23 18:34:17 vultr misskey[8307]: ERR  1        [api]        Internal error occurred in ap/show: character with byte sequence 0xe5 0xae 0xb6 in encoding "UTF8" has no equivalent in enccoding "LATIN1"
May 23 18:34:17 vultr misskey[8307]: WARN 1        [api]        SLOW API CALL DETECTED: ap/show (6383.825785999885ms)
May 23 18:34:53 vultr misskey[8307]: INFO 1        [remote ap]        Creating the Person: https://bgme.me/users/bgme
May 23 18:34:53 vultr misskey[8307]: ERR  1        [remote ap]        QueryFailedError: character with byte sequence 0xf0 0x9f 0x92 0x99 in encoding "UTF8" has no equivalent in encoding "LATIN1"
May 23 18:34:53 vultr misskey[8307]: ERR  1        [api]        Internal error occurred in ap/show: character with byte sequence 0xf0 0x9f 0x92 0x99 in encoding "UTF8" has no equivalent in encoding "LATIN1"
May 23 18:34:53 vultr misskey[8307]: WARN 1        [api]        SLOW API CALL DETECTED: ap/show (1077.6249270001426ms)
```
If I diable `signToActivityPubGet`, it will throw 401 Unauthorized error instead of the errors above.
```
May 23 19:20:48 vultr misskey[11966]: ERR  1        [api]        Internal error occurred in ap/show: 401 Unauthorized
May 23 19:23:03 vultr misskey[11966]: ERR  1        [api]        Internal error occurred in ap/show: 401 Unauthorized
```
I have set client encoding to UTF8 for my postgres and restarted it. It seems all right when using `SHOW client_encoding`

```
postgres=# SHOW client_encoding;
 client_encoding 
-----------------
 UTF8
(1 row)

misskey=# SHOW client_encoding;
 client_encoding 
-----------------
 UTF8
(1 row)
```
However, when I tried to list the databases. Encodings seem not have changed (still LATIN1).
```
postgres=# \list
                             List of databases
   Name    |  Owner   | Encoding | Collate | Ctype |   Access privileges   
-----------+----------+----------+---------+-------+-----------------------
 misskey   | postgres | LATIN1   | en_US   | en_US | =Tc/postgres         +
           |          |          |         |       | postgres=CTc/postgres+
           |          |          |         |       | misskey=CTc/postgres
 postgres  | postgres | LATIN1   | en_US   | en_US | 
 template0 | postgres | LATIN1   | en_US   | en_US | =c/postgres          +
           |          |          |         |       | postgres=CTc/postgres
 template1 | postgres | LATIN1   | en_US   | en_US | =c/postgres          +
           |          |          |         |       | postgres=CTc/postgres
```

No errors occur when I was trying to follow pleroma users or mastodon users on instances that disabled secure mode.

Thanks for your help.

**Comment by Johann150** [comment]
I don't think it is correct that the encoding is `LATIN1`. Here is the output from my instance:
```
misskey=> \list
                                   List of databases
     Name     |  Owner   | Encoding |   Collate   |    Ctype    |   Access privileges
--------------+----------+----------+-------------+-------------+-----------------------
 misskey      | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 |
...
```
You will most likely have to change your database encoding to UTF8, which seems to be the issue. The `signToActivityPubGet` seems to have nothing to do with this since your instance seems to be able to fetch the content fine, otherwise it would not get the encoding errors.

**Comment by georgianachao** [comment]
> I don't think it is correct that the encoding is `LATIN1`. Here is the output from my instance:
> 
> ```
> misskey=> \list
>                                    List of databases
>      Name     |  Owner   | Encoding |   Collate   |    Ctype    |   Access privileges
> --------------+----------+----------+-------------+-------------+-----------------------
>  misskey      | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 |
> ...
> ```
> 
> You will most likely have to change your database encoding to UTF8, which seems to be the issue. The `signToActivityPubGet` seems to have nothing to do with this since your instance seems to be able to fetch the content fine, otherwise it would not get the encoding errors.

Thanks for your help! I solved it using methods provided in this thread https://stackoverflow.com/questions/13115692/encoding-utf8-does-not-match-locale-en-us-the-chosen-lc-ctype-setting-requires . Also, I saw you had updated the manual and specified that the database encoding should be UTF-8. Just to let you know that `create database misskey with encoding = 'UTF8';` did not work in my case (I have tried to reinstall misskey multiple times). It will throw a error message saying `encoding UTF8 does not match locale en_US; the chosen LC_CTYPE setting requires encoding LATIN1`. I tried so many different ways to change the encoding to UTF8 but the only one that worked is the one that linked above. May be a good idea to remind people to check their encodings when installing postgresql using` \list` before they continue on the setup process. It will be quite annoying if you build up your instance for a while and later realize there's some federation issues that have to do with the database (and changing the encoding was quite troublesome in my case). Thank you so much for your replies and I'll close this issue.
